resources/prosody-plugins/mod_muc_displayname.lua

```diff
@@ -34,15 +38,14 @@ function filter_stanza_out(stanza, session)
         end
 
         if occupant.role ~= 'moderator' and not joining_moderator_participants[occupant.bare_jid] then
-            local st_clone = st.clone(stanza);
-            st_clone:remove_children('nick', NICK_NS);
-            return st_clone;
+            return filter_identity_from_presence(stanza);
```

Comment:
This will lead to problems when this stanza is modified for some users, but need to not be modified for others.
That is the reason of using a clone when modifying it.